frame/max-encoded-len/Cargo.toml

```diff
@@ -0,0 +1,35 @@
+[package]
+name = "max-encoded-len"
+version = "3.0.0"
+authors = ["Parity Technologies <admin@parity.io>"]
+edition = "2018"
+license = "Apache-2.0"
+homepage = "https://substrate.dev"
+repository = "https://github.com/paritytech/substrate/"
+description = "Trait MaxEncodedLen bounds the max encoded length of an item."
+
+
+[dependencies]
+codec = { package = "parity-scale-codec", version = "2.0.0", default-features = false }
+impl-trait-for-tuples = "0.2.1"
+max-encoded-len-derive = { package = "max-encoded-len-derive", version = "3.0.0", path = "derive", default-features = false, optional = true }
+primitive-types = { version = "0.9.0", default-features = false, features = ["codec"] }
+
+[dev-dependencies]
+codec = { package = "parity-scale-codec", version = "2.0.0", default-features = false, features = [ "derive" ] }
+rustversion = "1.0.4"
+trybuild = "1.0.42"
+
+[features]
+default = [
+	"derive",
+	"std",
+]
+derive = [
+	"max-encoded-len-derive",
+]
+std = [
+	"codec/std",
+	"max-encoded-len-derive/std",
+	"primitive-types/std",
```

Comment:
(note: once max encoded len is in parity-scale-codec then it should be the other way around, as codec shouldn't depend on primitive-types, but it is ok for now)